lms/djangoapps/grades/new/course_grade.py

```diff
@@ -24,8 +26,40 @@ class CourseGrade(object):
     def __init__(self, student, course, course_structure):
         self.student = student
         self.course = course
+        self.course_version = getattr(course, 'course_version', None)
+        self.course_edited_timestamp = getattr(course, 'subtree_edited_on', None)
         self.course_structure = course_structure
         self.chapter_grades = []
+        self._percent = None
+        self._letter_grade = None
+
+    @classmethod
+    def load_persisted_grade(cls, user, course, course_structure, current_grading_policy_hash):
+        """
+        Initializes a CourseGrade object, filling its members with persisted values from the database.
+
+        If the grading policy is out of date, recomputes the grade.
+
+        If no persisted values are found, returns None.
+        """
+        try:
+            persistent_grade = PersistentCourseGrade.read_course_grade(user.id, course.id)
+        except PersistentCourseGrade.DoesNotExist:
+            return None
+        course_grade = CourseGrade(user, course, course_structure)
+
+        if current_grading_policy_hash != persistent_grade.grading_policy_hash:
+            course_grade.compute_and_update(read_only=False)
+        else:
+            course_grade.percent = persistent_grade.percent_grade
+            course_grade.letter_grade = persistent_grade.letter_grade
+            course_grade.course_version = persistent_grade.course_version
+            course_grade.course_edited_timestamp = persistent_grade.course_edited_timestamp
+
+        course_grade._populate_course_structure_grades()
```

Comment:
This line shouldn't be called here - instead, just return fast access to the persisted course-only grade info.